components/atom/switch/src/index.js

```diff
@@ -16,108 +16,65 @@ export const TYPES = {
   SINGLE: 'single'
 }
 
+const SUPPORTED_KEYS = [' ', 'Enter', 'Spacebar']
+
 class AtomSwitch extends Component {
   state = {
-    toggle: false,
+    isToggle: this.props.initialValue,
     isFocus: false
   }
 
-  constructor(props) {
-    super(props)
+  _onKeyDown = event => {
+    if (this.props.disabled === true) return
 
-    this.executeIfEnabledFocusSwitch = this.executeIfEnabled(this.focusSwitch)
-    this.executeIfEnabledFocusOutSwitch = this.executeIfEnabled(
-      this.focusOutSwitch
-    )
-    this.executeIfEnabledToggleSwitch = this.executeIfEnabled(this.toggleSwitch)
-    this.executeIfEnabledActivateSwitch = this.executeIfEnabled(
-      this.activateToggle
-    )
-    this.executeIfEnabledDeactivateSwitch = this.executeIfEnabled(
-      this.deactivateToggle
-    )
-  }
-
-  componentDidMount() {
-    window.addEventListener('keydown', this.keyBindings, true)
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('keydown', this.keyBindings, true)
-  }
-
-  keyBindings = event => {
-    const {isFocus} = this.state
-    if (!isFocus || event.defaultPrevented) {
-      return
-    }
-    if (
-      event.key === 'Enter' ||
-      event.key === ' ' ||
-      event.key === 'Spacebar'
-    ) {
-      this.toggleSwitch()
+    if (SUPPORTED_KEYS.includes(event.key)) {
+      this._onToggle()
       event.preventDefault()
     }
   }
 
-  toggleSwitch = () => {
-    const {toggle} = this.state
+  _onToggle = forceValue => {
+    const {isToggle: stateToggle} = this.state
     const {onToggle} = this.props
-    this.setState({toggle: !toggle}, () => onToggle(!toggle))
-  }
-
-  activateToggle = () => {
-    this.setState({toggle: true})
+    const isToggle = forceValue !== undefined ? forceValue : !stateToggle
```

Comment:
why not `const isToggle = forceValue || !stateToggle`?

Reply:
Because `forceValue` could be `false` or `true`. If I use your code, when I try to force a value of `false` it will use always the `!stateToggle`